server/lib/two-factor-authentication.js

```diff
@@ -0,0 +1,80 @@
+import { get, isNil } from 'lodash';
+import speakeasy from 'speakeasy';
+
+import cache from './cache';
+import { crypto } from './encryption';
+
+export const ROLLING_LIMIT_CACHE_VALIDITY = 3600; // 1h in secs for cache to expire
+
+/**
+ * Verifies a TOTP against a user's 2FA token saved in the DB
+ * encryptedTwoFactorAuthToken = token saved for a User in the DB
+ * twoFactorAuthenticatorCode = 6-digit TOTP
+ */
+export function verifyTwoFactorAuthenticatorCode(encryptedTwoFactorAuthToken, twoFactorAuthenticatorCode) {
+  const decryptedTwoFactorAuthToken = crypto.decrypt(encryptedTwoFactorAuthToken);
+  const verified = speakeasy.totp.verify({
+    secret: decryptedTwoFactorAuthToken,
+    encoding: 'base32',
+    token: twoFactorAuthenticatorCode,
+    window: 2,
+  });
+  return verified;
+}
+
+export async function handleTwoFactorAuthenticationPayoutLimit(user, twoFactorAuthenticatorCode, expense) {
+  if (user.twoFactorAuthToken !== null) {
+    const host = expense.collective.getHostCollective();
+    const hostPayoutTwoFactorAuthenticationRollingLimit = get(
+      host,
+      'settings.payoutsTwoFactorAuth.rollingLimit',
+      1000000,
```

Comment:
The PR says that the default is 1,000, here it's 10,000. Not sure which one is right.

Reply:
The default for our basic "if payment is > $1000, trigger 2FA".

On the issue https://github.com/opencollective/opencollective/issues/3516 we say rolling limit of $10,000.